### PR TITLE
CORE-18533: Remove Checkpoint Retry

### PR DESCRIPTION
**Pull request by jujoramos**
- Remove "RetryState" avro schema.
- Move "FlowStates" enumeration to its own avro schema definition file.
- Remove "RETRYING" and set "FAILED" as the default value (for schema
  evolution, unmatched values do not throw an error but are resolved 
  to the default instead).
- Remove "retryState" and "maxFlowSleepDuration" fields from the
  "PipelineState" avro schema.
- Remove "CheckpointSchemaCompatibilityTest" (we only support evolving
  types one minor version at a time).
- Add schema compatibility tests for "FlowStates" and "PipelineState".

Runtime Change: https://github.com/corda/corda-runtime-os/pull/5374

### CODE REVIEW

**Review by simon-johnson-r3 — DISMISSED** [review]
LGTM

File: data/config-schema/src/main/resources/net/corda/schema/configuration/flow/1.0/corda.flow.json [hunk 1]
```diff
@@ -33,7 +33,7 @@
           "default": 16000
         },
         "maxFlowSleepDuration": {
-          "description": "The maximum delay in milliseconds before Corda schedules a periodic wake-up.",
+          "description": "Deprecated. The maximum delay in milliseconds before Corda schedules a periodic wake-up.",
```
Comment (LWogan) [inline data/config-schema/src/main/resources/net/corda/schema/configuration/flow/1.0/corda.flow.json:36]: Is this the official way we are to deprecate configs? Just curious as i know some other ones which need deprecating.

**Review by LWogan — APPROVED** [review]
LGTM

File: data/config-schema/src/main/resources/net/corda/schema/configuration/flow/1.0/corda.flow.json (hunk 1 above)
Comment (jujoramos) [reply data/config-schema/src/main/resources/net/corda/schema/configuration/flow/1.0/corda.flow.json:36]: I'm not 100% sure to be honest... @simon-johnson-r3, thoughts?

**Review by thiagoviana — APPROVED** [review]

**Pull request merged**